images/macos/software-report/SoftwareReport.Generator.ps1

```diff
@@ -22,322 +22,256 @@ Import-Module "$PSScriptRoot/../helpers/Xcode.Helpers.psm1"
 if (-not $os.IsBigSur) {
-    $utilitiesList += @(
-        (Get-VagrantVersion),
-        (Get-VirtualBoxVersion)
-    )
+    $utilities.AddToolNode("Vagrant", $(Get-VagrantVersion))
+    $utilities.AddToolNode("VirtualBox", $(Get-VirtualBoxVersion))
 }
-
-$markdown += New-MDList -Style Unordered -Lines ($utilitiesList | Sort-Object)
+$utilities.AddToolNode("yq", $(Get-YqVersion))
+$utilities.AddToolNode("zstd", $(Get-ZstdVersion))
 
 # Tools
-$markdown += New-MDHeader "Tools" -Level 3
-$toolsList = @(
-    (Get-JazzyVersion),
-    (Get-FastlaneVersion),
-    (Get-CmakeVersion),
-    (Get-AppCenterCLIVersion),
-    (Get-AzureCLIVersion),
-    (Get-AzureDevopsVersion),
-    (Get-AWSCLIVersion),
-    (Get-AWSSAMCLIVersion),
-    (Get-AWSSessionManagerCLIVersion)
-)
-
-if (-not $os.IsCatalina) {
-    $toolsList += @(
-        (Get-CodeQLBundleVersion)
-    )
-}
-
+$tools = $installedSoftware.AddHeaderNode("Tools")
 if ($os.IsLessThanMonterey) {
-    $toolsList += @(
-        (Get-AliyunCLIVersion)
-    )
+    $tools.AddToolNode("Aliyun CLI", $(Get-AliyunCLIVersion))
 }
-
-$toolsList += @(
-    (Get-XcodeCommandLineToolsVersion),
-    (Get-SwigVersion),
-    (Get-BicepVersion),
-    (Get-GHCupVersion),
-    (Get-GHCVersion),
-    (Get-CabalVersion),
-    (Get-StackVersion),
-    (Get-SwiftFormatVersion)
-)
-
+$tools.AddToolNode("App Center CLI", $(Get-AppCenterCLIVersion))
+$tools.AddToolNode("AWS CLI", $(Get-AWSCLIVersion))
+$tools.AddToolNode("AWS SAM CLI", $(Get-AWSSAMCLIVersion))
+$tools.AddToolNode("AWS Session Manager CLI", $(Get-AWSSessionManagerCLIVersion))
+$tools.AddToolNode("Azure CLI (azure-devops)", $(Get-AzureDevopsVersion))
+$tools.AddToolNode("Azure CLI", $(Get-AzureCLIVersion))
+$tools.AddToolNode("Bicep CLI", $(Get-BicepVersion))
+$tools.AddToolNode("Cabal", $(Get-CabalVersion))
+$tools.AddToolNode("Cmake", $(Get-CmakeVersion))
 if (-not $os.IsCatalina) {
-    $toolsList += @(
-        (Get-ColimaVersion)
-    )
+    $tools.AddToolNode("CodeQL Action Bundle", $(Get-CodeQLBundleVersion))
 }
-
-$markdown += New-MDList -Style Unordered -Lines ($toolsList | Sort-Object)
+if (-not $os.IsCatalina) {
+    $tools.AddToolNode("Colima", $(Get-ColimaVersion))
+}
+$tools.AddToolNode("Fastlane", $(Get-FastlaneVersion))
+$tools.AddToolNode("GHC", $(Get-GHCVersion))
+$tools.AddToolNode("GHCup", $(Get-GHCupVersion))
+$tools.AddToolNode("Jazzy", $(Get-JazzyVersion))
+$tools.AddToolNode("Stack", $(Get-StackVersion))
+$tools.AddToolNode("SwiftFormat", $(Get-SwiftFormatVersion))
+$tools.AddToolNode("Swig", $(Get-SwigVersion))
+$tools.AddToolNode("Xcode Command Line Tools", $(Get-XcodeCommandLineToolsVersion))
 
 # Linters
-$markdown += New-MDHeader "Linters" -Level 3
-$lintersList = @(
-    (Get-YamllintVersion),
-    (Get-SwiftLintVersion)
-)
+$linters = $installedSoftware.AddHeaderNode("Linters")
+$linters.AddToolNode("Yamllint", $(Get-YamllintVersion))
+$linters.AddToolNode("Swift", $(Get-SwiftLintVersion))

```

Comment:
Since we removed sorting, we need to swap these two items to follow alphabetical order.